perl/lib/Gherkin/AstBuilder.pm

```diff
@@ -60,7 +60,7 @@ sub build {
             Cucumber::Messages::Comment->new(
                 location => $self->get_location($token),
                 text     => $token->matched_text
-            );
+            ) unless $self->{uri} =~ m/\.md$/;
```

Comment:
This looks weird. Using Javascript as the reference implementation, I can't find this functionality.

Reply:
> I don't fully understand the Javascript implementation.

Well that makes two of us. :sweat: 

It looks intentional that table separators are matched as comments.

https://github.com/cucumber/gherkin/blob/214337104f9b110e5460c61528b760dfe93c5fbb/javascript/src/GherkinInMarkdownTokenMatcher.ts#L129C1-L136C4

But I can't work out how they're kept out of the AST.  And Unfortunately I don't have the time available for a real deep dive into this.  I hope Erik can provide some comments

Reply:
The naming of the function is misleading: Instead of matching it as a comment, it's actually matching it as "nothing at all", which is also why it's not ending up in the AST.